### PR TITLE
Feature/directory

### PR DESCRIPTION
**Pull request by ndickerson**
Load and Delete From Directory, along with increased unit test coverage and full documentation

### CODE REVIEW

File: src/main/java/com/bullhorn/dataloader/meta/Entity.java [hunk 1]
```diff
@@ -1,74 +1,103 @@
...
 	/**
 	 * @return the entity name usable in the Bullhorn's SDK-REST
 	 */
 	public String getEntityName() {
 		return entityInfo.getName();
 	}
-	
+
+	/**
+	 * @return the entity name usable in the Bullhorn's SDK-REST
```
Comment (acrowe2) [inline src/main/java/com/bullhorn/dataloader/meta/Entity.java]: copy/paste comment?  doesn't look like this is what the function actually does

Comment (ndickerson) [inline src/main/java/com/bullhorn/dataloader/meta/Entity.java]: Good catch - fixing!

File: src/main/java/com/bullhorn/dataloader/service/AbstractService.java [hunk 2]
```diff
@@ -110,37 +119,161 @@ protected ConcurrencyService createConcurrencyService(Command command, String en
...
     /**
+     * Given a file or directory, this method will determine all valid CSV files that can be used by DataLoader and
+     * collect them into a list indexed by the entity that they correspond to based on the filename. For a filename
+     * argument, this lits will be either empty or contain exactly one matching entity to filename.
+     *
+     * @param filePath any file or directory
+     * @param comparator
+     * @return a Map of entity enums to lists of valid files.
+     */
+    protected SortedMap<Entity, List<String>> getValidCsvFilesFromPath(String filePath, Comparator<Entity> comparator) {
```
Comment (acrowe2) [inline src/main/java/com/bullhorn/dataloader/service/AbstractService.java]: possibly break this up into smaller methods?

**Comment by ndickerson** [comment]
@acrowe2 - When reading from a directory, we have the need to prompt the user to Press the Enter key to accept the files that will be loaded prior to loading multiple files. We could just use `System.in` where we need it in AbstractService, but if we do, then we can't properly test it. So adding it as a Dependency Injected variable allows us to test it and declare dependencies. Plus, if we have a need for user input on the command line beyond this, we're already setup to do so.

File: src/main/java/com/bullhorn/dataloader/service/AbstractService.java (hunk 2 above)
Comment (ndickerson) [inline src/main/java/com/bullhorn/dataloader/service/AbstractService.java]: Refactored the method to pull out into separate smaller methods

**Pull request merged**